core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -580,7 +581,8 @@ object ConsumerGroupCommand extends Logging {
             groupId,
             Option(consumerGroup.coordinator),
             unassignedPartitions.keySet.toSeq,
-            unassignedPartitions.map { case (tp, offset) => tp -> Some(offset.offset) },
+            // Because there could be cases where the last commited offset is a negative integer (which translates to null [KAFKA-9507 for reference]), the following map function could lead to a NullPointerException. Two possible types are possible: OffsetsAndMetadata or null (which gets translated to None to avoid further exceptions)
+            unassignedPartitions.map { case (tp, offset) => tp -> ( if(offset.isInstanceOf[OffsetAndMetadata]) Some(offset.offset) else None ) },
```

Comment:
I am still not a fan of using `offset.isInstanceOf[OffsetAndMetadata]` in this case...

Now that we need the same logic twice, what about defining a small helper function which takes an `OffsetAndMetadata` and return its offset if not null?

```
// The admin client returns `null` as a value to indicate that there is not committed offset for a partition. 
def getPartitionOffset(tp: TopicPartition): Option[Long] = committedOffsets.get(tp).filter(_ != null).map(_.offset)
```

We could use it in both places.

Reply:
Absolutely right, same helper function works both cases and keeps the code cleaner and consistent. Going to change that.